### DISCUSSION TITLE
Quick query on the libraries in the lib folder

### DISCUSSION THREAD

**Discussion by TheShermanTanker** [msys2/MINGW-packages#11594] [Q&A]
Continued from https://github.com/msys2/MINGW-packages/issues/11593

Thanks Biswa and 1480 for the quick replies, seems like these archives are just names. Out of curiosity, given how these gcc ports have had the ability to link directly to a dll without import libraries for quite a while, does linking against these archives work similar to how the .lib files in Visual C++ do (to my knowledge MSVC does so by having a shim function inside the .lib that calls the real one in the dll) or do they work differently?

**Comment by Biswa96** [comment]
> to my knowledge MSVC does so by having a shim function inside the .lib that calls the real one

Those thunk functions are generated when the function declaration does not have `dllimport` attribute. But they are neither present in .lib or .dll.a file.
